tower-request-modifier/src/lib.rs

```diff
@@ -0,0 +1,191 @@
+extern crate futures;
+extern crate http;
+extern crate tower_service;
+
+use futures::Poll;
+use http::header::{HeaderName, HeaderValue};
+use http::uri::{self, Uri};
+use http::{HttpTryFrom, Request};
+use std::sync::Arc;
+use tower_service::Service;
+
+/// Wraps an HTTP service, injecting authority and scheme on every request.
+#[derive(Clone)]
+pub struct RequestModifier<T, B> {
+    inner: T,
+    modifiers: Arc<Vec<Box<dyn Fn(Request<B>) -> Request<B> + Send + Sync>>>,
+}
+
+impl<T, B> std::fmt::Debug for RequestModifier<T, B> {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> Result<(), std::fmt::Error> {
+        writeln!(f, "RequestModifier with {} modifiers", self.modifiers.len())
+    }
+}
+
+/// Configure an `RequestModifier` instance
+#[derive(Default)]
+pub struct Builder<B: Default> {
```

Comment:
I don't think the `B: Default` bound is needed. Instead, don't derive `Default`, but manually implement.